codec/src/main/java/io/netty/handler/codec/compression/CompressionCodecs.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import io.netty.util.internal.ObjectUtil;
+
+/**
+ * Creates a new {@link CompressionEncoder} or a new {@link CompressionDecoder}
+ * with specified {@link CompressionFormat}.
+ */
+public final class CompressionCodecs {
+
+    /**
+     * Factory method for a new {@link CompressionEncoder}.
+     * <p/>
+     * Note, that {@link CompressionFormat#ZLIB_OR_NONE} format is not allowed for an encoder.
+     *
+     * @param format specified compression format.
+     *
+     * @throws IllegalArgumentException if specified format is not supported.
+     */
+    public static CompressionEncoder newEncoder(CompressionFormat format) {
```

Comment:
Is it ok that factory returns high level interface? User has to cast it to `ClosableCompressionEncoder` if he wants to use additional methods.